src/ECS/Archetypes/CreatureArchetype.cpp

```diff
@@ -0,0 +1,35 @@
+/*****************************************************************************
+ * Copyright (c) 2018-2022 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "CreatureArchetype.h"
+
+#include <glm/gtx/euler_angles.hpp>
+
+#include "ECS/Components/Creature.h"
+#include "ECS/Components/Mesh.h"
+#include "ECS/Components/Transform.h"
+#include "ECS/Registry.h"
+#include "Enums.h"
+#include "Game.h"
+
+using namespace openblack;
+using namespace openblack::ecs::archetypes;
+using namespace openblack::ecs::components;
+
+entt::entity CreatureArchetype::Create(const std::string& playerName, CreatureType creatureType,
+                                       const std::string& creatureMind, const glm::vec3& position)
+{
+	auto& registry = Game::Instance()->GetEntityRegistry();
+	const auto entity = registry.Create();
+	auto meshId = creature::GetIdFromType(creatureType);
+	registry.Assign<Creature>(entity, playerName, creatureType, creatureMind);
+	registry.Assign<ecs::components::Mesh>(entity, meshId);
+	registry.Assign<Transform>(entity, position, glm::eulerAngleY(glm::radians(180.0f)), glm::vec3(.3f));
```

Comment:
Rotation and scale are magic values? Where do they come from?

Reply:
We use 180.0f elsewhere (StreetLanternArchetype, VillagerArchetype) to get meshes oriented correctly.
`.3f` gives the creature a reasonable size. 1.0f looks huge.

Reply:
What does vanilla do?

Reply:
Vanilla ignores the placement of the creatures and moves them to the temple. A good example of this is four gods where all A.I creatures are spawned at roughly the same location but are moved just afterwards to their pens.

![image](https://user-images.githubusercontent.com/32263167/186580646-aba3ddd0-04be-446a-aeb4-ad20e43625bb.png)

Reply:
And what scale do they start with?

Reply:
The scale is copied from player one when playing against the A.I otherwise other players' creatures.